### PR TITLE
Add default hash provider

### PR DESCRIPTION
**Pull request by x3medima17**
Signed-off-by: Dumitru <dimasavva17@gmail.com>

### Description of the Change
Extracted default hash provider from Signable

### Benefits
Now hash provider my be used not only by singable classes

### Possible Drawbacks 
None

### CODE REVIEW

File: shared_model/interfaces/base/signable.hpp [hunk 1]
```diff
@@ -20,6 +20,7 @@
 
 #include <boost/optional.hpp>
 
+#include "cryptography/default_hash_provider.hpp"
 #include "cryptography/hash_providers/sha3_256.hpp"
```
Comment (Alexey-N-Chernyshov) [inline shared_model/interfaces/base/signable.hpp]: seems should be removed

File: shared_model/interfaces/base/signable.hpp [hunk 2]
```diff
@@ -48,16 +49,16 @@ namespace shared_model {
...
-              typename HashProvider = shared_model::crypto::Sha3_256>
+              typename HashProvider = crypto::DefaultHashProvider>
     class Signable : public Primitive<Model, OldModel> {
 #else
     template <typename Model,
               typename HashProvider = shared_model::crypto::Sha3_256>
     class Signable : public ModelPrimitive<Model> {
 #endif
-     public:
       using HashProviderType = HashProvider;
```
Comment (Alexey-N-Chernyshov) [inline shared_model/interfaces/base/signable.hpp]: is it needed?

Comment (x3medima17) [reply shared_model/interfaces/base/signable.hpp]: If somebody passes another HashProvider - yes

File: shared_model/cryptography/default_hash_provider.hpp [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+#ifndef IROHA_DEFAULT_HASH_PROVIDER_HPP
+#define IROHA_DEFAULT_HASH_PROVIDER_HPP
+
+#include "cryptography/hash_providers/sha3_256.hpp"
+
+namespace shared_model {
+  namespace crypto {
+    using DefaultHashProvider = shared_model::crypto::Sha3_256;
```
Comment (muratovv) [inline shared_model/cryptography/default_hash_provider.hpp]: Add a description of the `using`.

File: shared_model/cryptography/default_hash_provider.hpp [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  }
```
Comment (muratovv) [inline shared_model/cryptography/default_hash_provider.hpp]: Looks like missing clang-format

**Pull request merged**